File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> numSet(nums.begin(), nums.end());
        int res = 0;
        for (int num : numSet){
            if (numSet.find(num - 1) == numSet.end()){
                int len = 1;
                while (numSet.find(num + len) != numSet.end()){
                    len++;
                }
                res = max(res, len);
            }
        }
        return res;
    }
};

File: Data Structures & Algorithms/count-subsequences/submission-2.py
class Solution:
    def numDistinct(self, s: str, t: str) -> int:
        if len(s) < len(t):
            return 0

        if s == t:
            return 1

        dp = [0] * (len(t) + 1)
        dp[-1] = 1

        for i in range(len(s) - 1, -1, -1):
            prev = 1
            for j in range(len(t) -1, -1, -1):
                res = dp[j] + (prev if s[i] == t[j] else 0)
                prev = dp[j]
                dp[j] = res

        return dp[0]